src/main.rs

```diff
@@ -0,0 +1,86 @@
+#![allow(non_camel_case_types)]
+
+#[macro_use]
+extern crate clap;
+extern crate crushtool;
+extern crate rustc_serialize;
+
+use std::fs::File;
+use std::io::{self, Error};
+use std::io::prelude::*;
+
+use rustc_serialize::json;
+
+use clap::{Arg, App};
+
+use crushtool::{decode_crushmap, encode_crushmap, CrushMap};
+// use crushtool::{CrushMap, BucketTypes, CrushBucketStraw, OpCode, BucketAlg, CrushRuleStep,
+//                 Bucket, CrushRuleMask, CrushHash, Rule, RuleType, CephVersion};
+// use crushtool::{CephCrushMap, CephDisk as Disk, CephHost as Host, CephPool as Pool, CephBucket,
+//                 CephBucketType};
+
+arg_enum!{
+  enum Mode {
+    compile,
+    decompile
+  }
+}
+
+fn main() {
+
+    let matches = App::new("crushtool")
+        .version(crate_version!())
+        .arg(Arg::with_name("mode")
+            .short("m")
+            .required(true)
+            .takes_value(true)
+            .help("Compile or decompile the crushmap")
+            .possible_values(&Mode::variants()))
+        .arg(Arg::with_name("custom")
+            .short("c")
+            .help("EXPERIMENTAL:: This will read in the custom crushmap syntax"))
+        .arg(Arg::with_name("output")
+            .short("o")
+            .help("Output file to put compiled crushmap into")
+            .required(true)
+            .takes_value(true))
+        .get_matches();
+
+
+    let mode = value_t!(matches.value_of("mode"), Mode).unwrap();
+
+    match mode {
+        Mode::compile => {
+            let mut input = String::new();
+            io::stdin().read_to_string(&mut input).unwrap();
```

Comment:
This is probably fine.  I doubt we'll ever get a crushmap input that is bigger than the heap size on the machine.